WordPress/gutenberg
# Interactivity API: Improve data-wp-context debugging by validating it as a stringified JSON Object.

## What?

Add a console warning if data-wp-context is not a JSON.

## Why?

I'm afraid it may be too late to prevent developers from adding any string into `data-wp-context`, but we can at least add a warning if the content of the directive is not the expected one.

## How?

I did on the `directives.js` but, should we add a `doing_it_wrong()` notice in the server instead?

## Testing Instructions




1.) Add an interactive block via `npx @wordpress/create-block@latest counter-block --template @wordpress/create-block-interactive-template`

2.) Edit the render file to use a non stringified JSON object as `data-wp-context`. Something like
```
// render.php
<div
	<?php echo get_block_wrapper_attributes(); ?>
	data-wp-interactive="create-block"
>
    <div data-wp-context="string" data-wp-on-document--scroll="actions.scroll1">
        Scroll 
        <br>
        <br>
        <br>
    </div>
</div>

```
3.) Add your block to a post or page.

4.) Check the warning on the frontend while running `npm run dev`.

5.) Check the warning does not appear if it has a production build.

## Screenshots or screencast 
![Screenshot 2024-04-24 at 17 44 19](https://github.com/WordPress/gutenberg/assets/37012961/c2dba2db-fca0-441a-9431-5a86031e6467)
